### ISSUE TITLE
ArrayIndexOutOfBoundsException in Android crash reporter

### ISSUE THREAD

**Issue by ascorbic** [microsoft/AppCenter-SDK-React-Native#568]
### **Description**

I have some devices that crash every time on launch with an ArrayIndexOutOfBoundsException while they try to submit an error report. I am using the React Native SDK, but the crash seems to be in the Android SDK code.

### **Repro Steps**

This is reproducible every time, but only on certain devices with a specific pending crash report. I am not sure how to find the pending report.

Note: this is a different device from the one with the bug fixed by #567 

### **Details**

1. Which SDK version are you using?
    - 1.13.0
2. Which OS version did you experience the issue on?
    - Android 6.0
3. What device version did you see this error on?  Were you using an emulator or a physical device?
    - Sony Xperia E5 physical device
4. What third party libraries are you using?
    - React Native 0.57.0
5. Please enable verbose logging for your app using `AppCenter.setLogLevel(Log.VERBOSE)` before your call to `AppCenter.start(...)` and include the logs here:

```
04-04 10:00:43.799  5358  5385 E AppCenter: Timeout waiting for looper tasks to complete.
04-04 10:00:43.799  5358  5385 E AndroidRuntime: FATAL EXCEPTION: AppCenter.Looper
04-04 10:00:43.799  5358  5385 E AndroidRuntime: Process: , PID: 5358
04-04 10:00:43.799  5358  5385 E AndroidRuntime: java.lang.ArrayIndexOutOfBoundsException: length=18; index=-8257536
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at java.util.ArrayList.get(ArrayList.java:310)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at java.io.ObjectInputStream.registeredObjectRead(ObjectInputStream.java:2120)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at java.io.ObjectInputStream.readCyclicReference(ObjectInputStream.java:807)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at java.io.ObjectInputStream.readNonPrimitiveContent(ObjectInputStream.java:773)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:1983)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:1940)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at java.io.ObjectInputStream.readFieldDescriptors(ObjectInputStream.java:909)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at java.io.ObjectInputStream.readClassDescriptor(ObjectInputStream.java:1722)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at java.io.ObjectInputStream.readNewClassDesc(ObjectInputStream.java:1634)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at java.io.ObjectInputStream.readClassDesc(ObjectInputStream.java:657)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at java.io.ObjectInputStream.readNewObject(ObjectInputStream.java:1782)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at java.io.ObjectInputStream.readNonPrimitiveContent(ObjectInputStream.java:761)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:1983)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:1940)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at java.io.ObjectInputStream.readNewArray(ObjectInputStream.java:1488)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at java.io.ObjectInputStream.readNonPrimitiveContent(ObjectInputStream.java:759)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:1983)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:1940)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at java.io.ObjectInputStream.readFieldValues(ObjectInputStream.java:1113)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at java.io.ObjectInputStream.defaultReadObject(ObjectInputStream.java:454)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at java.lang.Throwable.readObject(Throwable.java:440)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at java.io.ObjectInputStream.readObjectForClass(ObjectInputStream.java:1330)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at java.io.ObjectInputStream.readHierarchy(ObjectInputStream.java:1242)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at java.io.ObjectInputStream.readNewObject(ObjectInputStream.java:1835)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at java.io.ObjectInputStream.readNonPrimitiveContent(ObjectInputStream.java:761)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:1983)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:1940)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at com.microsoft.appcenter.utils.storage.FileManager.readObject(FileManager.java:162)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at com.microsoft.appcenter.crashes.Crashes.buildErrorReport(Crashes.java:774)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at com.microsoft.appcenter.crashes.Crashes.processPendingErrors(Crashes.java:667)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at com.microsoft.appcenter.crashes.Crashes.onStarted(Crashes.java:373)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at com.microsoft.appcenter.AppCenter.finishStartServices(AppCenter.java:898)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at com.microsoft.appcenter.AppCenter.access$600(AppCenter.java:51)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at com.microsoft.appcenter.AppCenter$8.run(AppCenter.java:820)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:815)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:104)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:207)
04-04 10:00:43.799  5358  5385 E AndroidRuntime: 	at android.os.HandlerThread.run(HandlerThread.java:61)
```

**Comment by MatkovIvan** [comment]
Hi @ascorbic,

Thanks for the report.
It looks like the file with the crash report is corrupted somehow and cannot be deserialized.

The SDK already catches [some already known](https://github.com/Microsoft/AppCenter-SDK-Android/blob/9ccb53463cfd81fefa4af4f316806e201936dedb/sdk/appcenter-crashes/src/main/java/com/microsoft/appcenter/crashes/Crashes.java#L773-L777) exceptions from [ObjectInputStream.readObject()](https://docs.oracle.com/javase/7/docs/api/java/io/ObjectInputStream.html#readObject()) Java API. So, we should probably generalize it a bit.


Thanks,
Ivan

**Comment by ascorbic** [comment]
Great. Thanks, @MatkovIvan Is there a way that I can access the crash report file on the device? I have several devices here which seem to be having similar issues and it might help to be able to see the broken reports.

**Comment by MatkovIvan** [comment]
The file that this code piece is trying to read is `<application dir>/error/*.throwable` (where application dir is usually `/data/data/<your package name>/files`)

**Comment by dhei** [comment]
Hi @ascorbic,

This bug should be fixed in v2.0.0 (https://github.com/microsoft/AppCenter-SDK-React-Native/releases/tag/2.0.0). Thanks.
